integration-tests/utils/integration-helpers.js

```diff
@@ -116,39 +118,75 @@ const makeGcpWorkspace = async (options) => {
   return await makeWorkspace({ ...options, hasBucket: true });
 };
 
-const deleteWorkspace = withSignedInPage(async ({ page, billingProject, workspaceName }) => {
-  try {
-    await page.evaluate(
-      async (name, billingProject) => {
-        await window.Ajax().Workspaces.workspace(billingProject, name).delete();
-      },
-      workspaceName,
-      billingProject
-    );
-    console.info(`Deleted workspace: ${workspaceName}`);
-  } catch (e) {
-    console.error(`Failed to delete workspace: ${workspaceName} with billing project: ${billingProject}`);
-    console.error(e);
-    throw e;
+const deleteWorkspaceInUi = async ({ page, billingProject, testUrl, workspaceName, retries = 5 }) => {
+  gotoPage(page, `${testUrl}#workspaces/${billingProject}/${workspaceName}`);
+  const isDeleted = await retryUntil({
+    getResult: async () => {
+      await findElement(page, clickable({ textContains: 'Action Menu' }, { timeout: Millis.ofMinute }));
+      await click(page, clickable({ textContains: 'Action Menu' }));
+      // Wait for popup to appear
+      await delay(Millis.of(300));
+      await findElement(page, clickable({ textContains: 'Delete' }));
+      await noSpinnersAfter(page, {
+        action: () => click(page, clickable({ textContains: 'Delete' })),
+      });
+      try {
+        await findText(page, "Please type 'Delete Workspace' to continue:");
+        return true;
+      } catch (e) {
+        await click(page, clickable({ textContains: 'Cancel' }));
+        return false;
+      }
+    },
+    interval: Millis.ofSeconds(30),
+    leading: true,
+    retries,
+  });
+  if (!isDeleted) {
+    return isDeleted;
   }
-});
+  await fillIn(page, input({ placeholder: 'Delete Workspace' }), 'Delete Workspace');
+  await noSpinnersAfter(page, {
+    action: () => click(page, clickable({ textContains: 'Delete workspace' })),
+    timeout: Millis.ofMinutes(2),
+  });
+  return true;
+};
 
-const deleteWorkspaceV2 = withSignedInPage(async ({ page, billingProject, workspaceName }) => {
+/** Deletes a workspace using the v2 delete endpoint. Individually deletes child resources if possible. Returns true if workspace deleted. */
+const deleteWorkspaceV2 = async ({ page, billingProject, workspaceName }) => {
   try {
-    await page.evaluate(
-      async (name, billingProject) => {
-        await window.Ajax().Workspaces.workspaceV2(billingProject, name).delete();
-      },
-      workspaceName,
-      billingProject
-    );
-    console.info(`Deleted workspace: ${workspaceName}`);
+    const isAppsEmpty = await deleteAppsV2({ page, billingProject, workspaceName });
+    const isRuntimesEmpty = await deleteRuntimesV2({ page, billingProject, workspaceName });
+    if (isAppsEmpty && isRuntimesEmpty) {
```

Comment:
I think its best practice to explicitly coerce this to a boolean here to avoid unintended javascript fun, `if (!!isAppsEmpty && !!isRuntimesEmpty)`. I would love if these files could be in typescript but I won't open that can of worms on you this farm along 😬

Reply:
In this case, both these values are supplied by functions which do return Boolean values (Array::every)